Update whosnext.py

kpasma opened the pull request:
Implemented an tunable pool size of possible presenter. In essence it automates the number of days you are free from presenting after you have presented. This was previously done by hand

kpasma committed:
Update whosnext.py

Implemented an tunable pool size of possible presenter. In essence it automates the number of days you are free from presenting after you have presented. This was previously done by hand

moorepants commented:
If `presenters_per_meeting` is 2 (with current members) then the result is 14 and if it is 1 the result is 70. Wouldn't this mean that as we make it through the pool of members that near the end of the member list, we are really having a random choice?

kpasma committed:
Update whosnext.py such that tuning is done by the number of people in the pool

Previously the tuning parameter had no logical meaning. Higher ment more people in the pool of possible presenters. 
Now the number of possible presentors for the next meeting is the tuning parameter itself. It directly determines how long people are exempt from giving a presentation.

kpasma commented:
I changed the way the time for exemption of presentation is calculated. Instead of having a nonsensical tuning parameter, I made the parameter "the number of people who can be chosen for the next presentation" the logical parameter to tune. So now you increase or decrease the pool of choices.
Now if `presenter_per_meeting` is 2, the number of people to chose from (a.k.a. `people_in_pool`) is 4, and we have 8 (excluding MC) members, the result will be 14. If `presenter_per_meeting` is 1, then the result is 56.

The end result is that for each time we use whosnext there will always be 4 people to chose from (these are the 4 people who presented the longest time ago in the past). The choice will be weighted, but it will remain random, with the assumption that no dates in the `presentation` list are in the future, and that whosnext is run after the meeting
The last assumption can be easily fixed by adding `days_between_meetings - 1`  to `free_days_post_pres`. But I refrained from doing this, as I did not completely thought it through, and I don't want to spend to much time on it.

moorepants commented:
Sounds good!
